src/proof/proof-service.ts

```diff
@@ -494,7 +517,19 @@ export class ProofService implements IProofService {
       if (globalStateInfo.replacedAtTimestamp === 0n) {
         throw new Error(`state was replaced, but replaced time unknown`);
       }
-      return false;
+
+      const timeDiff =
+        Date.now() -
+        getDateFromUnixTimestamp(Number(globalStateInfo.replacedAtTimestamp)).getTime();
+
+      if (
+        timeDiff >
+        (opts?.acceptedStateTransitionDelay ?? PROTOCOL_CONSTANTS.DEFAULT_AUTH_VERIFY_OPTS)
+      ) {
+        throw new Error('global state is outdated');
+      }
+
+      return true;
```

Comment:
let's remove this return true
Generaly looks like this function throws or return true, could be `void`, but for now I guess it's ok to have just one `return true` at the end